src/diff/changes/categorizedChange.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { RuleCategory } from "../ruleSet";
+
+/**
+ * Holds node change that is categorized by rules
+ */
+export class CategorizedChange {
```

Comment:
We don't have an `UncategorizedChange` class (at lower levels than this class, we use "delta" instead). Maybe just call this `Change` or `BaseChange`? Either would be a lot shorter.

Reply:
This is change after rule is applied which is categorized- We decided to call it as CategorizedChange, the other option was ClassifiedChange